morgoth/base/static/js/containers/AddonsListContainer.js

```diff
@@ -13,13 +13,20 @@ function mapStateToProps(state) {
 function mapDispatchToProps(dispatch) {
   return {
     fetchAddons: () => {
-      dispatch(fetchAddons()).then(response => {
-        if (response.error) {
-          dispatch(fetchAddonsFailure(response.payload));
-        } else {
-          dispatch(fetchAddonsSuccess(response.payload));
-        }
-      });
+      dispatch(fetchAddons())
+        .then(response => {
+          response.payload.json().then(jsonData => {
+            const data = jsonData === '' ? {} : jsonData;
+            if (response.payload.ok) {
```

Comment:
I'd usually avoid a level of indentation and write this like:

```javascript
dispatch(fetchAddons())
.then(response => [response, response.json()])
.then(([response, jsonData]) => {
  const data = jsonData || '';
  ...
});
```

Though that isn't a lot better. This is one problem with the fetch API, unfortunately.